src/vs/base/browser/ui/centered/centeredViewLayout.ts

```diff
@@ -91,21 +103,51 @@ export class CenteredViewLayout implements IDisposable {
 		this.height = height;
 		if (this.splitView) {
 			this.splitView.layout(width);
-			if (!this.didLayout) {
-				this.resizeMargins();
+			if (!this.didLayout || this.centeredLayoutFixedWidth) {
+				this.resizeSplitViews();
 			}
 		} else {
 			this.view.layout(width, height, top, left);
 		}
+
 		this.didLayout = true;
 	}
 
-	private resizeMargins(): void {
+	private resizeSplitViews(): void {
 		if (!this.splitView) {
 			return;
 		}
-		this.splitView.resizeView(0, this.state.leftMarginRatio * this.width);
-		this.splitView.resizeView(2, this.state.rightMarginRatio * this.width);
+		if (this.centeredLayoutFixedWidth) {
+
+			const centerViewWidth = Math.min(this.width, this.state.targetWidth);
+			const marginWidthFloat = (this.width - centerViewWidth) / 2;
+			this.splitView.resizeView(0, Math.floor(marginWidthFloat));
+			this.splitView.resizeView(1, centerViewWidth);
+			this.splitView.resizeView(2, Math.ceil(marginWidthFloat));
```

Comment:
@joaomoreno accepting this PR, but I think there is something better we can do with split views here. I feel like we should be able to set LayoutPriority based on the setting and the split view should handle this, but LayoutPriority is not honored in proportional layouting. With centered view fixed width, I would expect the margins to have a high layout priority and the centered view priority to be low and that would be all that's needed.

Reply:
I agree. This should've been contributed at the splitview level in some other shape or form. It seems that when this setting is `true`, multiple layouts will happen in a row now, right?